src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1001,6 +1001,30 @@
      "CREATE INDEX idx_resource_events_timestamp ON resource_events(timestamp)"
      "ALTER TABLE resource_events ADD CONSTRAINT resource_events_report_id_fkey FOREIGN KEY (report_id) REFERENCES reports(id) ON DELETE CASCADE"))
 
+(defn drop-latest-reports
+  []
+  (sql/do-commands
+    (if (scf-utils/postgres?)
+      "SELECT c.name,c.deactivated,lr.report_id as latest_report INTO certnames_transform FROM certnames c LEFT OUTER JOIN latest_reports lr ON c.name=lr.certname"
+      "CREATE TABLE certnames_transform (name, deactivated, latest_report) AS (SELECT c.name,c.deactivated,lr.report_id as latest_report FROM certnames c LEFT OUTER JOIN latest_reports lr ON c.name=lr.certname) WITH DATA")
+    "ALTER TABLE edges DROP CONSTRAINT edges_certname_fkey"
+    "ALTER TABLE catalogs DROP CONSTRAINT catalogs_certname_fkey"
+    "ALTER TABLE factsets DROP CONSTRAINT factsets_certname_fk"
+    "ALTER TABLE reports DROP CONSTRAINT reports_certname_fkey"
+    "ALTER TABLE latest_reports DROP CONSTRAINT latest_reports_certname_fkey"
+    "DROP TABLE certnames"
```

Comment:
Add an id to certnames is so tempting here, but I'd avoid it for now, we don't want to boil the ocean.